### PR TITLE
[Keyboard] P60

### PR DESCRIPTION
**Pull request by p3lim**
## Description

Converted my old handwired baby from TMK to QMK.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/p60/rules.mk [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MCU = atmega32u4
+F_CPU = 16000000
+F_USB = $(F_CPU)
+ARCH = AVR8
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+OPT_DEFS += -DBOOTLOADER_SIZE=512
+
+BOOTLOADER = halfkay
+
```
Comment (zvecr) [inline keyboards/handwired/p60/rules.mk:9]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = halfkay

# Build Options
#   change yes to no to disable
#
```

**Review by zvecr — CHANGES_REQUESTED** [review]
All `.c` and `.h` files need to have gpl compatible headers added

File: keyboards/handwired/p60/config.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#pragma once
```
Comment (drashna) [inline keyboards/handwired/p60/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/handwired/p60/keymaps/p3lim/keymap.c [hunk 3]
```diff
@@ -0,0 +1,146 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/handwired/p60/keymaps/p3lim/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/handwired/p60/keymaps/p3lim/keymap.c [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+bool process_record_user(uint16_t keycode, keyrecord_t *record){
+	switch(keycode){
+		case C_ESC0: // layer 0
+			if(record->event.pressed){
+				if(get_mods() & MOD_MASK_SHIFT)
+					register_code(KC_GRAVE);
+				else
+					register_code(KC_ESCAPE);
+			} else {
+				if(get_mods() & MOD_MASK_SHIFT)
```
Comment (drashna) [inline keyboards/handwired/p60/keymaps/p3lim/keymap.c:85]: Only issue with this, is if you release shift before this key, you may get a stuck key.  

May be better to unregister both, or use a static bool to track the status, and use that instead.

File: keyboards/handwired/p60/p60.c [hunk 5]
```diff
@@ -0,0 +1 @@
+#include "p60.h"
```
Comment (drashna) [inline keyboards/handwired/p60/p60.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/handwired/p60/p60.h [hunk 6]
```diff
@@ -0,0 +1,17 @@
+#pragma once
```
Comment (drashna) [inline keyboards/handwired/p60/p60.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/p60/keymaps/p3lim/keymap.c (hunk 4 above)
Comment (p3lim) [reply keyboards/handwired/p60/keymaps/p3lim/keymap.c:101]: What exact pattern are you concerned with? I've been using this for a long time and never had a key get stuck, and I'm unable to try and get it stuck intentionally.

Comment (drashna) [reply keyboards/handwired/p60/keymaps/p3lim/keymap.c:101]: shift down, escape down, shift up, escape up.  

however, if you're fine with it, as is, then that's fine by me.

File: keyboards/handwired/p60/rules.mk [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+EXTRAKEY_ENABLE = yes
+CONSOLE_ENABLE = yes
+COMMAND_ENABLE = yes
+NKRO_ENABLE = yes
+
+SLEEP_LED_ENABLE = no
+BACKLIGHT_ENABLE = no
+RGBLIGHT_ENABLE = no
+BLUETOOTH_ENABLE = no
+AUDIO_ENABLE = no
```
Comment (fauxpark) [inline keyboards/handwired/p60/rules.mk:21]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/handwired/p60/readme.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
+# P60
+
+![](https://user-images.githubusercontent.com/26496/108102379-c3562d00-7088-11eb-9083-9921497429ca.JPG)
```
Comment (fauxpark) [inline keyboards/handwired/p60/readme.md:3]:
```suggestion
![](https://i.imgur.com/7AfyZkkl.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/p60/keymaps/p3lim/keymap.c (hunk 4 above)
Comment (p3lim) [reply keyboards/handwired/p60/keymaps/p3lim/keymap.c:101]: Ah, guess I've never accidentally hit that combination.

File: keyboards/handwired/p60/readme.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
+# P60
+
+[![](https://user-images.githubusercontent.com/26496/108596854-ee41c900-7387-11eb-8225-e4faf72aa793.png)](https://user-images.githubusercontent.com/26496/108102379-c3562d00-7088-11eb-9083-9921497429ca.JPG)
```
Comment (fauxpark) [inline keyboards/handwired/p60/readme.md:3]:
```suggestion
![](https://i.imgur.com/7AfyZkkl.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
Also, this board has no default keymap.

File: keyboards/handwired/p60/readme.md (hunk 9 above)
Comment (p3lim) [reply keyboards/handwired/p60/readme.md:3]: Doesn't matter which hosting service I use, GitHub is fine.

**Comment by p3lim** [comment]
A default keymap is pointless since this is a handwired board, and the personal keymap is simple enough to understand if someone really wants to remake my keyboard exactly.

File: keyboards/handwired/p60/readme.md (hunk 9 above)
Comment (fauxpark) [reply keyboards/handwired/p60/readme.md:3]: The image is way too big, though.

**Comment by fauxpark** [comment]
Certain parts of QMK rely on a default keymap existing.

File: keyboards/handwired/p60/readme.md (hunk 9 above)
Comment (p3lim) [reply keyboards/handwired/p60/readme.md:3]: The embedded image is the one you resized, it's linking to the full-size one. Look at the rich diff.

Comment (fauxpark) [reply keyboards/handwired/p60/readme.md:3]: It's *linking* to the thumbnail-sized one, but the image is still the full size.

**Pull request closed without merging**